internal/events/event.go

```diff
@@ -459,29 +465,30 @@ func (e Events) queryEvents(ctx context.Context, params *common.V2GetEventsParam
 
 	params.Limit, params.Offset = preparePaginationParams(params.Limit, params.Offset)
 	if *params.Limit == 0 {
-		return make([]*common.Event, 0), eventSeverityCount, nil
+		return make([]*common.Event, 0), eventSeverityCount, &eventCount, nil
 	}
 
 	err = tx.Offset(int(*params.Offset)).Limit(int(*params.Limit)).Find(&events).Error
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
-	return events, eventSeverityCount, nil
+	return events, eventSeverityCount, &eventCount, nil
```

Comment:
I think &eventCount should be changed to eventCount here. &eventCount will give you the address of the pointer, this is not what you want.

Proof:
https://go.dev/play/p/PGWEbbUj7r-

Reply:
No longer relevant